userspace/libsinsp/events/sinsp_events_ppm_sc.cpp

```diff
@@ -231,32 +231,50 @@ libsinsp::events::set<ppm_sc_code> libsinsp::events::sys_sc_set()
 	return sc_set;
 }
 
-libsinsp::events::set<ppm_sc_code> libsinsp::events::names_to_sc_set(const std::unordered_set<std::string>& syscalls)
+libsinsp::events::set<ppm_sc_code> libsinsp::events::event_names_to_sc_set(const std::unordered_set<std::string>& events)
+{
+	/* Convert event names into an event set, and then convert that into a
+	 * syscall set. We exclude generics due to the potential information loss
+	 * (e.g. one generic event will include all generic syscalls in the
+	 * conversion). Generics are handled below using their actuall syacall name.
+	 * Note: this is the same logic with which the "evt.type" filter field
+	 * is extracted. */
+	auto gen_event_set = libsinsp::events::set<ppm_event_code>(
+		{ PPME_GENERIC_E, PPME_GENERIC_X });
+	auto event_set = libsinsp::events::names_to_event_set(events);
```

Comment:
maybe this API should be called `event_names_to_event_set`?

Reply:
Since we are here I would also rename these 2:
```
std::unordered_set<std::string> event_set_to_names(const set<ppm_event_code>& events_set, bool resolve_generic = true);

/*!
  \brief Get the ppm_event of all the event names provided in the set.
*/
set<ppm_event_code> names_to_event_set(const std::unordered_set<std::string>& events);
```

into `event_set_to_event_names` and `event_names_to_event_set`

We can do that also in another PR, no strong opinion here :)

Reply:
I would agree, however for the event set I don't see a case in which we'll ever have a `sc_names_to_event_set`. My proposal is to move this discussion to another issue or PR, WDYT?

Reply:
agree with that :)